src/sentry/services/hybrid_cloud/audit/impl.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from typing import Any, Mapping
+
+from sentry.models import AuditLogEntry, AuthIdentity, AuthProvider, Organization, Team, User
+from sentry.services.hybrid_cloud.audit import AuditLogMetadata, AuditLogService
+from sentry.services.hybrid_cloud.auth import ApiAuthIdentity, ApiAuthProvider
+from sentry.services.hybrid_cloud.organization import ApiOrganizationMember
+from sentry.services.hybrid_cloud.user import APIUser, user_service
+
+
+class DatabaseBackedAuditLogService(AuditLogService):
+    @staticmethod
+    def _resolve_user(user: APIUser | int | None) -> User | None:
+        if user is None:
+            return None
+        if isinstance(user, int):
+            user_id = user
+        else:
+            user_id = user.id
+        return User.objects.get(id=user_id)
+
+    def write_audit_log(self, *, metadata: AuditLogMetadata, data: Mapping[str, Any]) -> None:
+        organization = Organization.objects.get(id=metadata.organization.id)
```

Comment:
How is this working?  I'm pretty sure as written, this database backed service must run on `control`, which means this shouldn't be able to fetch an organization object.

Would it also not suffice to simple capture the organization id and pass it to the `create`?

Basically, we cannot realistically have an Organization ORM lookup in the same logic that writes an AuditLogEntry.

Reply:
That's a mistake on my part. I believe it would have shown up running `AuthHelperTest` with `stable=True`, if it weren't being masked by a different availability error that's already coming from `ApiInviteHelper`. I'll be fixing it as part of getting those tests to pass. In the meantime, we can treat it like https://github.com/getsentry/sentry/blob/09b79a9a645b716b6d41a1b06b43b8225b80e777/src/sentry/services/hybrid_cloud/auth/impl.py#L272-L274